src/next-edge/index.ts

```diff
@@ -100,6 +114,12 @@ export function createApiHandler(options: CreateApiHandlerOptions) {
       (req as unknown as { secure: boolean }).secure ||
       req.headers['x-forwarded-proto'] === 'https'
 
+    for (const [name] of Object.entries(req.headers)) {
+      if (!forwardedHeaders.includes(name)) {
+        delete req.headers[name]
+      }
+    }
```

Comment:
I think

```suggestion
    req.headers = Object.fromEntries(
        Object.entries(req.headers)
            .filter(([key]) => forwardedHeaders.includes(key))
```

is more readable, but that is up for debate I guess :shrug:

Reply:
Sure I guess :smile: